torch/nn/functional.py

```diff
@@ -3004,3 +3014,29 @@ def fold(input, output_size, kernel_size, dilation=1, padding=0, stride=1):
         raise NotImplementedError("Input Error: Only 3D input Tensors are supported (got {}D)".format(input.dim()))
         ret = input  # TODO: remove when jit supports exception control flow
     return ret
+
+
+@weak_script
+def pad_circular(input, padding):
```

Comment:
question: why didn't we add this functionality to `F.pad`?

Reply:
Hi @fmassa! That was a suggestion to have padding functionally at one place.
cc: @soumith , @ezyang